src/VirtualClient/VirtualClient.Actions/Furmark/FurmarkExecutor.cs

```diff
@@ -0,0 +1,257 @@
+
+            string commandArguments = $"-accepteula -s -i {this.SessionId} -w {this.packageDirectory} {this.ExecutableLocation} /width={this.Width} /height={this.Height} /msaa={this.Msaa} /max_time={this.Time} /nogui /nomenubar /noscore /run_mode=1 /log_score /disable_catalyst_warning /log_temperature /max_frames";
+  
+            using (IProcessProxy process = await this.ExecuteCommandAsync(this.PSexecExecutablePath, commandArguments, this.packageDirectory, telemetryContext, cancellationToken, runElevated: false))
+            {
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    if (process.StandardError.Length > 0) 
+                    {
+                        Console.WriteLine($"The resultfilepath is = '{this.ResultsFilePath}' ");
+
+                        string[] outputFilePaths = new string[] { $"{this.ResultsFilePath}", $"{this.XMLFilePath}" };
+                        IEnumerable<string> results = await this.LoadResultsAsync(outputFilePaths, CancellationToken.None);
+
+                        await this.LogProcessDetailsAsync(process, telemetryContext, "Furmark", logToFile: true);
+                        await this.CaptureMetricsAsync(process, this.ResultsFilePath, telemetryContext, cancellationToken);
+                        await this.CaptureMetricsAsync(process, this.XMLFilePath, telemetryContext, cancellationToken);
+
+                    }
+                    else
+                    {
+                        string[] outputFilePaths = new string[] { $"{this.ResultsFilePath}", $"{this.XMLFilePath}" };
+                        IEnumerable<string> results = await this.LoadResultsAsync(outputFilePaths, CancellationToken.None);
+
+                        await this.LogProcessDetailsAsync(process, telemetryContext, "Furmark", logToFile: true);
+                        process.ThrowIfWorkloadFailed();
+                    }
+
+                }
+            }
+ 
+        }
+
+        private async Task CaptureMetricsAsync(IProcessProxy process, string resultsFilePath, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                if (!this.fileSystem.File.Exists(resultsFilePath))
+                {
+                    throw new WorkloadResultsException(
+                        $"The Furmark results file was not found at path '{resultsFilePath}'.",
+                        ErrorReason.WorkloadFailed);
+                }
+
+                string results = await this.LoadResultsAsync(resultsFilePath, cancellationToken);
+
+                await this.LogProcessDetailsAsync(process, telemetryContext, "Furmark", results.AsArray(), logToFile: true);
+
+                IList<Metric> metrics;
+                if (resultsFilePath == this.XMLFilePath)
+                {
+                    FurmarkMetricsParser2 furmarkParser = new FurmarkMetricsParser2(results);
```

Comment:
Can we have name of parser in such a way that we can differentiate between both parsers? like FurmarkXmlMetricsParser, something like that?